### ISSUE TITLE
[BUG] Offset in API does not working correctly 

### ISSUE THREAD

**Issue by DarKDinDoN** [vigoren/foundryvtt-simple-calendar#145]
**Describe the bug**
Related to this bugfix: https://github.com/vigoren/foundryvtt-simple-calendar/issues/133 (which is now fixed and confirmed)
The api doesn't return the correct day with an offset.

**To Reproduce**
1. Same configuration as in the previous ticket
2. Check the result of timestampToDate function

✅ OK
![image](https://user-images.githubusercontent.com/1687854/123397928-46d9da80-d5a3-11eb-8b4a-38319f90af39.png)
![image](https://user-images.githubusercontent.com/1687854/123398040-62dd7c00-d5a3-11eb-9823-4e90b9f2ebd3.png)

✅ OK
![image](https://user-images.githubusercontent.com/1687854/123398081-6ec93e00-d5a3-11eb-8530-2b362f79ecbb.png)
![image](https://user-images.githubusercontent.com/1687854/123398115-7852a600-d5a3-11eb-8e6d-aa3e2c06c73e.png)

✅ OK
![image](https://user-images.githubusercontent.com/1687854/123398181-87395880-d5a3-11eb-9d6d-a0c066051a16.png)
![image](https://user-images.githubusercontent.com/1687854/123398222-8ef8fd00-d5a3-11eb-8e82-f0d9f9ded6f2.png)

❎ NOK (should be 18)
![image](https://user-images.githubusercontent.com/1687854/123398274-9c15ec00-d5a3-11eb-969f-52187c896643.png)
![image](https://user-images.githubusercontent.com/1687854/123398315-a637ea80-d5a3-11eb-828c-542126e718a9.png)

**Comment by vigoren** [comment]
This was a bug but not as you were expecting.

The day is the index of that day for that month, since Mois du Père is split into 2 months (with an intercalary month in between) the 18th should be 0 (since it is the first day of that month) the 19th would be 1 etc.

This was from a miscalculation on my part, that I have fixed (will be apart of the next release).

To help with displaying the correct date I added 2 new properties to the timestampToDate return
- dayOffset: Which is the offset for the days of that month
- dayDisplay: Which is how the day is displayed on the calendar

So using dayDisplay would show the correct day in anything using this function to display a date.

So for your last example the returned result would look like this
```
{
    day: 1, // index for days in this month, 19 is the second day of this month so an index of 1
    dayDisplay: '19', // How the day is shown on the calendar
    dayOffset: 17, // How many days this month days are offset from the "normal" count
    month: 6, // The month was actually being advanced too far in some cases so this will align it back correctly
    ... // All other values will be unchanged
}
```
